joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/AttributeAccessorTests.scala

```diff
@@ -0,0 +1,50 @@
+package io.joern.rubysrc2cpg.querying
+
+import io.joern.x2cpg.Defines
+import io.joern.rubysrc2cpg.testfixtures.RubyCode2CpgFixture
+import io.shiftleft.codepropertygraph.generated.{Operators, DispatchTypes}
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, FieldIdentifier, Identifier}
+import io.shiftleft.semanticcpg.language.*
+
+class AttributeAccessorTests extends RubyCode2CpgFixture {
+
+  "`x.y=1` is approximated by a `x.y =` assignment with argument `1`" in {
+    val cpg = code("""x = Foo.new
+                     |x.y = 1
+                     |""".stripMargin)
+    inside(cpg.assignment.where(_.source.isLiteral.codeExact("1")).l) {
+      case xyAssign :: Nil =>
+        xyAssign.lineNumber shouldBe Some(2)
+        xyAssign.code shouldBe "x.y = 1"
+
+        val fieldTarget = xyAssign.target.asInstanceOf[Call]
+        fieldTarget.code shouldBe "x.@y"
+        fieldTarget.name shouldBe Operators.fieldAccess
+        fieldTarget.methodFullName shouldBe Operators.fieldAccess
+        fieldTarget.dispatchType shouldBe DispatchTypes.STATIC_DISPATCH
+
+        inside(fieldTarget.argument.l) {
+          case (base: Identifier) :: (field: FieldIdentifier) :: Nil =>
+            base.name shouldBe "x"
+            field.canonicalName shouldBe "@y"
+          case xs => fail("Expected field access to have two targets")
+        }
+      case xs => fail("Expected a single assignment to the literal `1`")
+    }
+  }
+
+  "`x.y` is represented by a CALL `x.y`" in {
+    val cpg = code("""x = Foo.new
+        |x.y
```

Comment:
What happens for `a.b = x.y` ?
In that case the `x.y` needs to be a field access. Can you please write a test for that?

Reply:
My concern is that this is not true in all cases. e.g. 
```ruby
class Foo {
  def a
     puts "hello"
  end
}
x = Foo.new
y = SomeBar.new
y.b = x.a
```
In this case, `x.a` is in-fact a call, and not referring to the member access

Reply:
Thats a valid concern but without this we wont have solved the issue of requiring to track back to the creation of x in order to learn about the function a.
Lets for now represent `x.a` as a field access if there are no parenthesis.
Its likely that we will change that later, but it is the better start.